setup.py

```diff
@@ -33,14 +33,19 @@
     base_dir = Path(__file__).parent
     src_dir = base_dir / "src"
 
-    about = {}
+    about: dict[str, str] = {}
     with (src_dir / "layered_config_tree" / "__about__.py").open() as f:
         exec(f.read(), about)
 
     with (base_dir / "README.rst").open() as f:
         long_description = f.read()
 
-    install_requirements = ["pyyaml>=5.1"]
+    install_requirements = [
+        "pyyaml>=5.1",
+        # typing extensions
+        "types-PyYAML",
```

Comment:
Should these be in a separate `dev` install block? You wouldn't need them if you were going to install this library in an environment and then use it in a notebook, for example, but I'm not sure how it would work if vivarium which uses layered_config_tree had mypy run on it.